packages/driver/src/cy/multi-domain/index.ts

```diff
@@ -144,56 +129,138 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
         Cypress.action('cy:enqueue:command', attrs)
       }
 
-      const updateCommand = (details) => {
-        if (details.logAdded) {
-          const attrs = details.logAdded
+      const updateCommand = ({ id, end }) => {
+        if (end) {
+          const command = commands[id]
 
-          attrs.consoleProps = () => details.logAdded.consoleProps
-          attrs.renderProps = () => details.logAdded.renderProps
+          if (command) {
+            delete commands[id]
+            command.deferred.resolve()
+          }
+        }
+      }
 
-          const log = Cypress.log(attrs)
+      const onLogAdded = ({ logAdded }) => {
+        const attrs = logAdded
 
-          logs[log.get('id')] = log
+        attrs.consoleProps = () => logAdded.consoleProps
+        attrs.renderProps = () => logAdded.renderProps
 
-          return
+        const log = Cypress.log(attrs)
+
+        // if the log needs to stream updates, defer its result to make sure all streamed updates come in
+        if (!attrs.ended) {
+          logs[log.get('id')] = {
+            log,
+            deferred: createDeferred(),
+          }
         }
+      }
+
+      const onLogChanged = ({ logChanged }) => {
+        const { deferred, log } = logs[logChanged.id]
 
-        if (details.logChanged) {
-          const log = logs[details.logChanged.id]
+        // NOTE: Cypress.LogConfig only contains partial types of what exists on the log attributes, missing a lot of 'private' properties
+        const logAttrs = log.get()
 
-          if (log) {
-            log.set(details.logChanged)
+        const updatedLogAttributes: Partial<Cypress.Log> = difference(logChanged, logAttrs)
```

Comment:
Is a deep difference truly necessary? When it gets merged into the log with `log.set()`, it does a shallow merge with `_.extend`, so any properties deeper than the second level will be overwritten on the log anyway.

Reply:
Probably not. Sounds like we just need to check one level deep for an object to see if there is a difference, and if so just return the different object that is keyed, as opposed to recursively trying to merge it. Will take a look and update it!

Reply:
wound up just doing something like below and getting rid of the `difference` util all together
```js
      const updatedLogAttributes: Partial<Cypress.Log> = _.transform(logChanged, (result: {[key: string]: any }, value: any, key: string) => {
          if (!_.isEqual(value, logAttrs[key])) {
            result[key] = value
          }
        })
``